episodes/3-monitor-the-model.Rmd

```diff
@@ -82,65 +82,29 @@ Index(['DATE', 'MONTH', 'BASEL_cloud_cover', 'BASEL_humidity',
        'TOURS_temp_mean', 'TOURS_temp_min', 'TOURS_temp_max'],
       dtype='object')
 ```
+There is a total of 9 different measured variables (global_readation, humidity, etcetera)
 
-::: challenge
-## Exercise: Explore the dataset
-Let's get a quick idea of the dataset.
-
-* How many data points do we have?
-* How many features does the data have (don't count month and date as a feature)?
-* What are the different types of measurements (humidity etc.) in the data and how many are there?
-* (Optional) Plot the amount of sunshine hours in Basel over the course of a year. Are there any interesting properties that you notice?
 
-:::: solution
-
-## Solution
+Let's have a look at the shape of the dataset:
 ```python
 data.shape
 ```
-This will give both the number of datapoints (3654) and the number of features (89 + month +
-te).
-To see what type of features the data contains we could run something like:
-```python
-import string
-print({x.lstrip(string.ascii_uppercase + "_") for x in data.columns if x not in ["MONTH", "DATE"]})
-```
 ```output
-{'cloud_cover', 'precipitation', 'sunshine', 'global_radiation', 'temp_mean', 'humidity', 'pressure', 'temp_min', 'temp_max'}
-```
-An alternative way which is slightly more complicated but gives better results is using regex.
-```python
-import re
-feature_names = set()
-for col in data.columns:
-   feature_names.update(re.findall('[^A-Z]{2,}', col))
-feature_names
+(3654, 91)
 ```
-In total there are 9 different measured variables.
-
-### Optional exercise
-You can plot the sunshine hours in Basel as follows:
-```python
-data.iloc[:365]['BASEL_sunshine'].plot(xlabel="Day",ylabel="Basel sunchine hours")
-```
-
-![](../fig/03_exploration_basel_sunshine_graph.png){alt='Plot of Basel sunshine hours in one year'}
-
-There are a couple of things that might stand out to you. For example, it looks like the sunshine
-hours are fluctuating a lot per day. There also seems to be seasonal fluctuation, with the peaks
-becoming higher around the middle of the year.
-::::
-:::
+This will give both the number of samples (3654) and the number of features (89 + month +
+date).
 
 ## 3. Prepare data
 
 ### Select a subset and split into data (X) and labels (y)
 The full dataset comprises of 10 years (3654 days) from which we will select only the first 3 years. The present dataset is sorted by "DATE", so for each row `i` in the table we can pick a corresponding feature and location from row `i+1` that we later want to predict with our model. As outlined in step 1, we would like to predict the sunshine hours for the location: BASEL.
 
 ```python
-nr_rows = 365*3
+nr_rows = 365*3 # 3 years
 # data
-X_data = data.loc[:nr_rows].drop(columns=['DATE', 'MONTH'])
+X_data = data.loc[:nr_rows] # Select first 3 years
+X_data = X_data.drop(columns=['DATE', 'MONTH']) # Drop date and month column
```

Comment:
I think this makes it a bit clearer what we are doing here. It can be a lot to explain if it's just a oneliner.